runtime/src/ancient_append_vecs.rs

```diff
@@ -314,6 +318,28 @@ impl AccountsDb {
         );
     }
 
+    /// for each account in `unrefed_pubkeys`, in each `accounts_to_combine`, addref
+    fn addref_accounts_failed_to_shrink_ancient(&self, accounts_to_combine: AccountsToCombine) {
+        self.thread_pool_clean.install(|| {
+            accounts_to_combine
+                .accounts_to_combine
+                .into_par_iter()
+                .for_each(|combine| {
+                    self.accounts_index.scan(
```

Comment:
I expect these entries to all be in-mem since we've modified the refcounts. What we're parallel scanning is this:
1. we iterated the storages that were eligible for packing
2. each dead account in those storages was unref'd in the index, with the expectation we'd be rewriting/packing/shrinking the alive accounts.
So, it is these sets of dead accounts (one set per storage we originally scanned) that we unref'd that we're scanning in parallel, one set per slot that was eligible/chosen for packing. It is perfectly legal and reasonable to parallel scan the index. It is binned into many buckets, with each bucket having its own independent locks and such.
Also, we don't expect this code to run in reality. This is just required to correctly clean up when we are unable to complete the packing process that should 100% of the time succeed. The shrink code today has a similar failing condition. The condition here for failure is that we have N required slots to store packed accounts in, but there are not that many slots in the set we're considering where ALL accounts in the slot have only 1 refcount.